### ISSUE TITLE
appstream-util rejects MIT as a metdata license.

### ISSUE THREAD

**Issue by aleixpol** [hughsie/appstream-glib#320]
As can be seen here:
https://github.com/flathub/org.qgis.qgis/issues/35
https://flathub.org/builds/#/builders/48/builds/812

This should work though, MIT it's even one of the suggested values in the documentation:
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

**Comment by hughsie** [comment]
@ximion added the MIT licence as acceptable in the specification, but I never agreed the MIT license with the Red Hat lawyers, we only agreed on the subset that appstream-glib supports. I can chase them, but legal stuff takes time.

**Comment by ximion** [comment]
Jup, I told @aleixpol that story already.
The problem is that the specification from the very start mentioned "user permissive licenses" implying that all permissive licenses are acceptable, while also listing a bunch as examples, including MIT.
I never didn't know that appstream-glib was restricting the list for a really long time, and I am very reluctant to remove a licenses that appstream accepted since forever from the supported set.

**Comment by hughsie** [comment]
Also, a question from RH legal: "Which MIT license" -- looking at https://spdx.org/licenses/ there are a few to choose from...

**Comment by ximion** [comment]
Since the identifier that is checked for is `MIT`, it's https://spdx.org/licenses/MIT.html and maybe also https://spdx.org/licenses/MIT-0.html (if the former is good, the latter must from my perspective be fine as well).

**Comment by hughsie** [comment]
So the proposal to legal would be both `MIT` and `MIT-0` as well?

**Comment by ximion** [comment]
Yes, I think that's sensible - if `MIT` is permitted, people may want the no-attribution `MIT-0` as well, and since that one's basically `MIT` with one restriction removed, it should also work for metadata collections if the `MIT` license itself is considered to be fine for that purpose.

**Comment by ximion** [comment]
@hughsie From looking at the patch above, it looks like libappstream also accepts BSL-1.0 as valid. I think that's fair game as the Boost License is basically a "public domain" license for countries which don't have the concept of public domain, and by its text is more permissive than BSD and MIT: https://en.wikipedia.org/wiki/Boost_(C%2B%2B_libraries)#License
But tbh I completely forgot that that license was also permitted...

**Comment by scx** [comment]
See also: https://github.com/flathub/org.freedesktop.appstream-glib/issues/6#issuecomment-532904944
